src/qtt/algorithms/generic.py

```diff
@@ -581,10 +585,47 @@ def weightedCentroid(im, contours, contourIdx, fig=None):
         plt.plot(xyw[0], xyw[1], '.g', markersize=17)
     return xyw
 
+
+def boxcar_filter(signal, kernel_size):
+    """ Perform boxcar filtering on an array.
+    At the edges, the edge value is replicated beyond the edge as needed by the size of the kernel.
+    This is the 'nearest' mode of scipy.ndimage.convolve. For details, see
+    https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.convolve.html?highlight=mode
+
+
+    Args:
+        signal (sequence): a sequence containing the signal to be filtered.
+        kernel_size (tuple): Multidimensional size of the filter box. Must have the same number of dimensions as the signal.
+
+    Returns:
+        a numpy array containing the filtered signal.
+    """
+
+    if not isinstance(signal, np.ndarray):
+        signal = np.array(signal)
+
+    if len(kernel_size) != len(signal.shape):
+        raise RuntimeError('Number of dimensions of kernel (%d) not equal to number of dimension of input signal (%d)' %
+                           (len(kernel_size), len(signal.shape)))
+
+    for siz in kernel_size:
```

Comment:
Replace the next two lines with:
```
if np.any(kernel_size<=0):
   raise ...
```

Reply:
Ah, those fancy numpy functions

Reply:
Just need to convert the kernel_size to a np.array first